### PR TITLE
Contact Info form

### PR DESCRIPTION
**Pull request by dalex135**
@Bharathravi1995 
Please check whether the app runs without an error.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "zone.js": "~0.10.2"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.901.4",
+    "@angular-devkit/build-angular": "^0.901.5",
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/package.json]: Why did u include ^ with .5 patch version? I hope ~ with .4 patch version itself be able install latest minor patch version.

File: frontend/src/app/forms/forms-routing.module.ts [hunk 2]
```diff
@@ -16,12 +6,13 @@ import { FormsPage } from './forms.page';
...
-  },
+    component: FormsPage
+
+  }
 ];
 
 @NgModule({
   imports: [RouterModule.forChild(routes)],
   exports: [RouterModule],
 })
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/forms-routing.module.ts]: Why did you change this file? Seems just removed comments. Just revert this changes.

File: frontend/src/app/forms/side-menu/side-menu.component.html [hunk 3]
```diff
@@ -10,15 +15,38 @@ <h1>Progress</h1>
...
+
+          <ion-icon [name]="menu.isCompleted? 'checkmark-circle-outline' : 'ellipse-outline' "
+            [color]="menu.isActive?'tertiary' : 'medium'"></ion-icon>
+          <ion-label class="ion-padding-horizontal" [color]="menu.isActive?'tertiary' : 'medium'">{{menu.displayName}}
+          </ion-label>
+        </ion-item>
+      </ion-menu-toggle>
+
+    </ion-list>
+
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/side-menu/side-menu.component.html]: Resolve the conflict in this file and push. I hope there won't be any changes in this file for you.

File: frontend/src/app/forms/side-menu/side-menu.component.scss [hunk 4]
```diff
@@ -1,3 +1,9 @@
+<<<<<<< HEAD
+.font25{
+    font-size:25px;
+}
+=======
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/side-menu/side-menu.component.scss]: Resolve the conflict in this file and push. I hope there won't be any changes in this file for you.

File: frontend/src/app/forms/side-menu/side-menu.component.ts [hunk 5]
```diff
@@ -17,6 +20,19 @@ import { Component, EventEmitter, Input, Output } from '@angular/core';
   styleUrls: ['./side-menu.component.scss'],
 })
 export class SideMenuComponent {
+<<<<<<< HEAD
+
+  @Input() sideMenuList: Array<{ id: number, displayName: string, isCompleted: boolean, isActive: boolean, selector: string }>
+
+  @Output() private pageProgressStatusCallBack = new EventEmitter();
+
+  constructor() { }
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/side-menu/side-menu.component.ts]: Resolve the conflict in this file and push. I hope there won't be any changes in this file for you.

File: frontend/src/app/forms/steps/contact-info/contact-info.component.html [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<ion-list>
+  <ion-item>
+    <br />
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/steps/contact-info/contact-info.component.html:3]: We should avoid br tag instead try to handle with css

File: frontend/src/app/forms/steps/contact-info/contact-info.component.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      (ionChange)="onValChange()"
+    ></ion-input>
+  </ion-item>
+
+  <ion-item>
+    <ion-label position="floating" color="medium">LinkedIn Account</ion-label>
+    <ion-input
+      type="url"
+      [formControl]="contactInfoObject.controls.linkedIn"
+      (ionChange)="onValChange()"
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/steps/contact-info/contact-info.component.html:28]: As suggested by Martin we have to move (ionChange) . Instead we can use observers.

File: frontend/src/app/forms/steps/contact-info/contact-info.component.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+})
+export class ContactInfoComponent implements OnInit {
+  @Input() menuObject: {
+    id: number;
+    displayName: string;
+    isCompleted: boolean;
+    isActive: boolean;
+    selecteor: string;
+  };
+  @Input() contactInfoObject: any;
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/steps/contact-info/contact-info.component.ts:16]: any type should not be used. Linter should have thrown. Run linter and prettier by running non run all-checks command. Fix and push the code

File: frontend/src/app/forms/steps/contact-info/contact-info.component.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+  @Output() pageProgressStatusCallBack = new EventEmitter();
+
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/forms/steps/contact-info/contact-info.component.ts:22]: Empty methods to be removed. Will enable lint for this.

File: frontend/src/app/home/home.page.html [hunk 10]
```diff
@@ -13,16 +21,21 @@
           d="M393.062,55.031h14.282V69.125H393.062V55.031Z"
           transform="translate(-375.5, -55)"
         ></path>
+>>>>>>> a40cdd678f8684ffbf0ed13799cb2e793f90f684
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/home/home.page.html]: Resolve conflicts. I don't think u need to change anything in this file.

File: frontend/src/app/home/home.page.scss [hunk 11]
```diff
@@ -24,6 +24,25 @@
...
+  color: var(--ion-color-secondary-contrast);
+  background: var(--ion-color-secondary);
+  padding: 1rem;
+  border-radius: 1000px 0 1000px 1000px;
+  font-weight: 500;
+  letter-spacing: 4px;
+  width: 100%;
+  outline: none;
+
+  // Add some spacing so that the button is not "pressed" against the above content when the screen has a small height.
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/home/home.page.scss]: Resolve conflict by using incoming change sets. CSS in this file has been moved.

File: frontend/src/app/home/home.page.ts [hunk 12]
```diff
@@ -7,9 +7,18 @@ import { Router } from '@angular/router';
   styleUrls: ['home.page.scss'],
 })
 export class HomePage {
+<<<<<<< HEAD
+
+  constructor(private router: Router) { }
+
+  onStartApplication(): void {
+    this.router.navigate(['/forms'], { queryParams: { step: 1 } });
+
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/app/home/home.page.ts]: Resolve conflict and push

File: frontend/src/global.scss [hunk 13]
```diff
@@ -27,6 +27,8 @@
 
 @import './theme/custom.scss';
 @import './theme/fonts.scss';
+<<<<<<< HEAD
+=======
```
Comment (Bharathwaj-Ravi-1995) [inline frontend/src/global.scss]: Resolve conflict and push

**Review by Bharathwaj-Ravi-1995 — COMMENTED** [review]
Pipeline has to pass for this PR. Also there are many conflicts needs to be resolved. While resolving choose the incoming change sets if u have not made any changes.

**Comment by MBuchalik** [comment]
In case you get this working some day: I would highly recommend that you combine your commits into one. The commit history looks awful :D

My suggestion is even to start with a fresh branch: Create a new branch that is up-to-date. Then manually incorporate your changes. And then push the changes. I can assist with this if you like.

**Comment by georg-schwarz** [comment]
@MBuchalik pls leave commit history in there since it is one part of the grading...

**Comment by MBuchalik** [comment]
> I would highly recommend that you combine your commits into one.

> @MBuchalik pls leave commit history in there since it is one part of the grading...

Alright 👍 I thought that with this particular case (35+ commits) the author should simply (on his own) squash all into one. Wouldn't have changed the content or anything, but made his own history more clean. But let's leave it as it is :)

**Comment by dalex135** [comment]
@Bharathravi1995 
I pushed the working code to my branch. Please let me know whether it is visible for you?
If not I can create a new pull request.

**Comment by dalex135** [comment]
Will be merging from a different branch
